### ISSUE TITLE
Pulseaudio crackling sound 

### ISSUE THREAD

**Issue by Gfurst** [godotengine/godot#17676]
**Godot version:**

happens in godot tag  2.1.4
and 2.1 at 0b55481ceaf3dc90c03bf19e0cf9db9d72dd8893

**OS/device including version:**

Linux guidesk 4.15.3-2-ARCH #1 SMP PREEMPT Thu Feb 15 00:13:49 UTC 2018 x86_64 GNU/Linux

**Issue description:**

Building with `pulseaudio=yes` enabled, results with the pulseaudio server craclking audio issues, immediately noticeable right after opening the editor. 

**Steps to reproduce:**

Just compiling with PA yes/no

**Notes:**

Possible correlation with issues #15266 and #14866. I personally suspect it being related to the forced latency (which is also present with cpu usage), or that "silent" issue that could have solved it on 3.0.

I can run diagnostics commands with `pacmd`, but I haven't seen anything out of the ordinary yet, feel free to ask me anything I can do. Running `godot -v`:

```
 AudioServerSW Params: 
 -mix chans: 2
 -mix rate: 44100
 -latency: 10
 -fx: 1
 -interp: 1
```

**Comment by akien-mga** [comment]
CC @marcelofg55 

I can't reproduce it by just running the compiled binary, but I did experience such random crackling issues playing [Deep Sixed](http://store.steampowered.com/app/591000/Deep_Sixed/) which uses Godot 2.1.4. (Also [confirmed by another Linux player](https://itch.io/t/199540/sound-has-random-noise)).

**Comment by Gfurst** [comment]
more info:
* pulseaudio ver: 11.1
* card name: "HDA ATI SB"
* card driver: "snd_hda_intel"

One thing I've noticed is a huge drop in latency as soon as godot starts:
```
pacmd list-sinks | grep current\ latency
current latency: 355,03 ms
current latency: 1,38 ms
current latency: 2,82 ms
```
First was before opening godot, second is with pulse enable and third with it disabled. No issues with it disabled, so it may be completely irrelevant.

**Comment by marcelofg55** [comment]
Did you try this also with latest Godot master (aka 3.x)?
Also could you run godot with -verbose flag? And paste here the console output, thanks!

**Comment by Gfurst** [comment]
no I did not, yet at least, and I mentioned above that one possible closed issue may be the answer, the verbose output is plain and simple, without errors with crackling right from the get go

```
./godot.x11.opt.tools.64.llvm.pulse.180321 -v     
No touch devices found
AudioServerSW Params: 
 -mix chans: 2
 -mix rate: 44100
 -latency: 10
 -fx: 1
 -interp: 1
Using GLES2 video driver
GLES2: Using GLEW 1.13.0
remap failed: /home/guiu/.godot/editor_settings.tres
load resource: /home/guiu/.godot/editor_settings.tres
EditorSettings: Load OK!
EditorSettings Save OK!
FPS: 1
FPS: 53
FPS: 59
FPS: 60
```

**Comment by Calinou** [comment]
@Gfurst Can you (or anyone else) still reproduce this bug in Godot 3.2.1 or [3.2.2beta4](https://godotengine.org/article/dev-snapshot-godot-3-2-2-beta-4)?

Also, note that there's an OS-independent bug report about clipping/static noises in rapid intervals of sound: https://github.com/godotengine/godot/issues/22016

**Comment by Calinou** [comment]
Closing due to lack of response. Please comment if you can still reproduce this bug.

PS: If no [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) was included in the original bug report, please upload one as well to ease troubleshooting.

**Comment by Gfurst** [comment]
Hello @Calinou, I haven't checked with this for a while, and this was an issue back when using Godot2 (which had to do because of my project backward compatibility), I can try a newer release to check this out, but I presume it isn't the default packaged released, but a custom build with pulseaudio option?

**Comment by Calinou** [comment]
@Gfurst The official releases use PulseAudio by default. You need to explicitly opt-in to use ALSA by specifying `--audio-driver ALSA` on the Godot command line.

**Comment by Gfurst** [comment]
Well the original issue was rather old, so at some point in time, things must've changed. Anyway, I can still try that out, but if it was the default for everyone, I doubt I would be the only one having issues.

**Comment by akien-mga** [comment]
There were some issues in 2.1 but they have long since been fixed.

**Comment by aaaaaaaaargh** [comment]
Sorry, but this issues does not seem to be resolved yet.

I am experiencing the same behaviour with the recent godot-mono build (3.2.3-1) found in the AUR repository of Arch. This build is using the PA driver by default and apparently it is impossible to switch to ALSA using the command line parameter.

I just tested with various projects and regardless of which one it is I always end up with a very annoying crackling and stuttering on the whole system. I also tried to use Jack and bridge to PA using a sink device which ended up in a lot of Xruns while Godot was running.

That being said, I think this issue needs to be reopened.
